uitools/import/Esri/ArcGISRuntime/Toolkit/ScaleLineImpl.qml

```diff
@@ -0,0 +1,85 @@
+import QtQuick 2.15
+import QtQml 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Shapes 1.15
+
+import Esri.ArcGISRuntime.Toolkit.Controller 100.13
+
+/*!
+  \internal
+  This is the display portion of a Scaleline. A single scaleline can be
+  composed of multiple display portions (i.e. showing imperial and metric scalelines side
+  by side).
+ */
+Control {
+    id: scaleLine
+
+    /*!
+      \qmlproperty ScalelineController controller
+      \brief The controller used for calculations based on the mapView.
+    */
+    property var controller: null
+
+    /*!
+      \qmlproperty unitSystem
+      \brief UnitSystem for the Scaleline. For this display portion,
+      only metric and imperial are valid values.
+    */
+    property int unitSystem
+
+    contentItem: Shape {
+        anchors.fill:parent
+        ShapePath {
+            strokeColor: "black"
+            strokeWidth: 3
+            fillColor: Qt.rgba(255, 255, 255, 0.5)
+            startX: 0; startY: 0
+            PathLine { x: 0; y: scaleLine.height }
+            PathLine { x: internal.displayWidth; y: scaleLine.height }
+            PathLine { x: internal.displayWidth; y: 0 }
+        }
+        Label {
+            anchors {
+                left: parent.left
+                verticalCenter: parent.verticalCenter
+                leftMargin: scaleLine.spacing
+            }
+            font: scaleLine.font
+            palette: scaleLine.palette
+            text: internal.displayDistance
+        }
+    }
+
+    Component.onCompleted: {
+        if (controller.mapView) {
+            internal.calculateScale();
+        }
+    }
+
+    QtObject {
+        id: internal
+
+        property real displayWidth
+        property string displayDistance
+
+        function calculateScale() {
+            // Get the total scalebar width in pixels.
+            const availableWidth = scaleLine.width;
+            // Get the possible longest distance the scalebar would cover.
+            const maxDistance = controller.calculateDistance(availableWidth);
+            // Select the most appropriate distance the scalebar should cover.
+            const displayDistance = controller.calculateBestScalebarLength(maxDistance);
+            // Calculate the new displayed scalebar width.
+            internal.displayWidth = controller.calculateDisplayWidth(displayDistance, maxDistance, availableWidth);
+            // Display the value of the calculated distance.
+            internal.displayDistance = controller.calculateDistanceInDisplayUnits(displayDistance, controller.unitSystem);
+        }
+
+        property Connections geoViewConnections: Connections {
+            target: controller.mapView
+            function onViewpointChanged() {
+                internal.calculateScale();
+            }
```

Comment:
This may be a design consideration, but viewpointChanged is very noisy. An alternate approach is to set the scalebar gray and remove the value while interacting (GeoView.drawStatus == DrawStatus::InProgress), and then finalize it once draw status is Completed. I think what you have is fine and we should only consider this if there are issues.

Another alternate approach would be to only recalculate this value if the map scale actually changed. So vertical panning of the view wouldn't cause the scale bar to constantly reevaluate. Not sure if that would work for the SceneView.

Reply:
Currently I'd err on the side of making the scalebar update when interacting. Keyboard interaction when one vertically pans the map updates the scalebar, and I'd possibly want to keep this sort of live update. 

Not to say it couldn't be turned off with a future configuration open. 

https://user-images.githubusercontent.com/42203545/132228190-1f617a3a-bee2-4214-832e-133b22ab559e.mp4